tests/unit/plugins/modules/packaging/language/test_npm.py

```diff
@@ -58,13 +118,54 @@ def test_absent(self):
             'state': 'absent'
         })
         self.module_main_command.side_effect = [
-            (0, '{"dependencies": {"coffee-script": {}}}', ''),
+            (0, '{"dependencies": {"coffee-script": {"version" : "2.5.1"}}}', ''),
             (0, '{}', ''),
         ]
 
         result = self.module_main(AnsibleExitJson)
 
         self.assertTrue(result['changed'])
         self.module_main_command.assert_has_calls([
+            call(['/testbin/npm', 'list', '--json', '--long', '--global'], check_rc=False, cwd=None),
+            call(['/testbin/npm', 'uninstall', '--global', 'coffee-script'], check_rc=True, cwd=None),
+        ])
+
+    def test_absent_version(self):
+        set_module_args({
+            'name': 'coffee-script',
+            'global': 'true',
+            'state': 'absent',
+            'version': '2.5.1'
+        })
+        self.module_main_command.side_effect = [
+            (0, '{"dependencies": {"coffee-script": {"version" : "2.5.1"}}}', ''),
+            (0, '{}', ''),
+        ]
+
+        result = self.module_main(AnsibleExitJson)
+
+        self.assertTrue(result['changed'])
+        self.module_main_command.assert_has_calls([
+            call(['/testbin/npm', 'list', '--json', '--long', '--global'], check_rc=False, cwd=None),
+            call(['/testbin/npm', 'uninstall', '--global', 'coffee-script'], check_rc=True, cwd=None),
+        ])
+
+    def test_absent_version_different(self):
+        set_module_args({
+            'name': 'coffee-script',
+            'global': 'true',
+            'state': 'absent',
+            'version': '2.5.1'
+        })
+        self.module_main_command.side_effect = [
+            (0, '{"dependencies": {"coffee-script": {"version" : "2.5.0"}}}', ''),
+            (0, '{}', ''),
+        ]
+
+        result = self.module_main(AnsibleExitJson)
+
+        self.assertTrue(result['changed'])
+        self.module_main_command.assert_has_calls([
+            call(['/testbin/npm', 'list', '--json', '--long', '--global'], check_rc=False, cwd=None),
             call(['/testbin/npm', 'uninstall', '--global', 'coffee-script'], check_rc=True, cwd=None),
```

Comment:
I might be misunderstanding the intention, but shouldn't `uninstall` have `name@version` in this case as version was specified?

This behavior is defined on line 167 of `npm.py`
```python
 if kwargs['version'] and self.state != 'absent':
```